js/gated-content/src/components/personal-training/PersonalTrainingListing.vue

```diff
@@ -0,0 +1,170 @@
+          :key="personalTraining.id"
+          :video="personalTraining"
+        />
+      </div>
+    </template>
+  </div>
+</template>
+
+<script>
+import client from '@/client';
+import Spinner from '@/components/Spinner.vue';
+import PersonalTrainingTeaser from '@/components/personal-training/PersonalTrainingTeaser.vue';
+import { JsonApiCombineMixin } from '@/mixins/JsonApiCombineMixin';
+import { FavoritesMixin } from '@/mixins/FavoritesMixin';
+import { ListingMixin } from '@/mixins/ListingMixin';
+
+export default {
+  name: 'PersonalTrainingListing',
+  mixins: [JsonApiCombineMixin, FavoritesMixin, ListingMixin],
+  components: {
+    PersonalTrainingTeaser,
+    Spinner,
+  },
+  props: {
+    title: {
+      type: String,
+      default: 'Personal trainings',
+    },
+    msg: {
+      type: String,
+      default: 'No personal trainings found.',
+    },
+    viewAll: {
+      type: Boolean,
+      default: false,
+    },
+    sort: {
+      type: Object,
+      default() {
+        return { path: 'date.value', direction: 'ASC' };
+      },
+    },
+    limit: {
+      type: Number,
+      default: 0,
+    },
+  },
+  data() {
+    return {
+      loading: true,
+      error: false,
+      listing: [],
+      links: {},
+      params: [
+        'instructor_id',
+      ],
+    };
+  },
+  watch: {
+    $route: 'load',
+    sort: 'load',
+  },
+  async mounted() {
+    await this.load();
+  },
+  computed: {
+    listingIsNotEmpty() {
+      return this.listing !== null && this.listing.length > 0;
+    },
+  },
+  methods: {
+    async load() {
+      this.loading = true;
+      const params = {};
+      if (this.params) {
+        params.include = this.params.join(',');
+      }
+
+      params.filter = {
+        dateFilter: {
+          condition: {
+            path: 'date.end_value',
+            operator: '>=',
+            value: new Date().toISOString(),
+          },
+        },
+        orGroup: {
+          group: {
+            conjunction: 'OR',
+          },
+        },
+        userFilter1: {
+          condition: {
+            path: 'customer_id.drupal_internal__uid',
+            operator: '=',
+            value: window.drupalSettings.user.uid,
+            memberOf: 'orGroup',
+          },
+        },
+        userFilter2: {
+          condition: {
+            path: 'instructor_id.drupal_internal__uid',
+            operator: '=',
+            value: window.drupalSettings.user.uid,
+            memberOf: 'orGroup',
+          },
+        },
+      };
+
+      params.sort = {
+        sortBy: this.sort,
+      };
+
+      if (this.favorites) {
+        if (this.isFavoritesTypeEmpty('personal_training', 'personal_training')) {
+          this.loading = false;
+          return;
+        }
+        params.filter.includeFavorites = {
+          condition: {
+            path: 'drupal_internal__id',
+            operator: 'IN',
+            value: this.getFavoritesTypeIds('personal_training', 'personal_training'),
```

Comment:
weird - we are passing same string twice?

Reply:
That is a method of this component that receive's 2 variables - entity_type, bundle